build-js.sh

```diff
@@ -73,876 +72,49 @@ OUT="$PWD/out"
 
 echo "generate embedded intrinsics module"
 
-python3 $BINARYEN_SCRIPTS/embedwast.py $BINARYEN_SRC/passes/wasm-intrinsics.wast $BINARYEN_SRC/passes/WasmIntrinsics.cpp
+python $BINARYEN_SCRIPTS/embedwast.py $BINARYEN_SRC/passes/wasm-intrinsics.wast $BINARYEN_SRC/passes/WasmIntrinsics.cpp
 
 echo "compiling source files"
 
-mkdir -p ${OUT}
+# internal source files or directories to exclude
+EXCLUDE=(
+  "$BINARYEN_SRC/support/archive.cpp"
+  "$BINARYEN_SRC/support/command-line.cpp"
+  "$BINARYEN_SRC/support/path.cpp"
+  "$BINARYEN_SRC/tools/"
+)
+
+# external source files to include
+INCLUDE=()
+
+mapfile -t SOURCES < <(ls $BINARYEN_SRC/**/*.cpp)
```

Comment:
How about using `find -name "*.cpp"`?

Reply:
Are you proposing the following?

```
mapfile -t SOURCES < <(find $BINARYEN_SRC -name "*.cpp")
```

Reply:
Yeah, I'm just more familiar with that than the `**` trick in ls.. not a huge deal.. as I say hopefully we can remove this completely and switch the cmake.